pkg/sql/information_schema.go

```diff
@@ -552,6 +588,25 @@ func forEachColumnInTable(
 	return nil
 }
 
+func forEachForeignKeyInTable(
+	table *sqlbase.TableDescriptor,
+	fn func(
+		*sqlbase.IndexDescriptor,
+		*sqlbase.ForeignKeyReference,
+	) error,
+) error {
+	for i := range table.Indexes {
```

Comment:
I'm a little sad that we're duplicating this logic again -- the goal of `TableDescriptor.GetConstraintInfo` was to return all the info needed to write something like this and avoid duplication